test/test_jit_cuda_fuser.py

```diff
@@ -42,6 +42,24 @@
 FUSION_GROUP = 'prim::CudaFusionGroup'
 FUSION_GUARD = 'prim::CudaFusionGuard'
 
+import contextlib
+
+@contextlib.contextmanager
+def nvfuser_singleton_fusion(flag):
+    old_value = torch._C._jit_set_nvfuser_single_node_mode(flag)
+    try:
+        yield
+    finally:
+        torch._C._jit_set_nvfuser_single_node_mode(old_value)
+
+@contextlib.contextmanager
+def nvfuser_horizontal_fusion(flag):
+    old_value = torch._C._jit_set_nvfuser_horizontal_mode(flag)
```

Comment:
Question: Why are there two switches?  If horizontal fusion is off, isn't that singleton fusion?

Reply:
So single node fusion just allows us to construct a fusion group with single node. op_authoring guys are asking that API for their test facilities.
horizontal fusion simply disables input sharing... This could WAR the PW scheduler issue that LTC guys are having.

e.g. with horizontal fusion turned off, we can still fuse a chain of operations, or branches joining together. Maybe there's a better name for that (sibling fusion?)